test/check-machines-migrate

```diff
@@ -0,0 +1,248 @@
+        machine1.execute("echo '10.111.113.1 machine1' >> /etc/hosts")
+        machine2.execute("echo '10.111.113.2 machine2' >> /etc/hosts")
+        machine2.execute("echo '10.111.113.1 machine1' >> /etc/hosts")
+
+        # Older libvirt versions require hostname != localhost else they result in error
+        # internal error hostname of destination resolved to localhost, but migration requires an FQDN
+        if machine1.image in ["rhel-8-5", "centos-8-stream", "fedora-33", "fedora-34"]:
+            machine2.execute("sed -i 's/127.0.1.1 /c\127.0.1.1 machine2 machine2/' /etc/hosts")
+            machine2.execute("hostnamectl set-hostname machine2")
+            machine2.execute("systemctl restart libvirtd")
+
+        # setup key authentication
+        machine1.execute("mkdir -m 0700 -p $HOME/.ssh")
+        machine1.execute("ssh-keygen -t rsa -N '\' -f $HOME/.ssh/id_rsa")
+        machine1.execute("echo 'StrictHostKeyChecking no' > $HOME/.ssh/config")
+        machine1PubKey = machine1.execute("cat $HOME/.ssh/id_rsa.pub")
+        machine2.execute("echo '{0}' >> /root/.ssh/authorized_keys".format(machine1PubKey))
+
+        # Setup the default storage pool on the destination host
+        # This is normally automatically created when creating the first VM but in this case our dest VM was never populated
+        xml = POOL_XML.format(path="/var/lib/libvirt/images")
+        machine2.execute("echo \"{0}\" > /tmp/xml && virsh pool-define /tmp/xml && virsh pool-start images".format(xml))
+
+        # Plain migration: The source host VM opens a direct unencrypted TCP connection to the destination host for sending the migration data.
+        # Unless a port is manually specified, libvirt will choose a migration port in the range 49152-49215, which will need to be open in the firewall on the remote host.
+        # https://wiki.libvirt.org/page/FAQ
+        machine2.execute("systemctl start firewalld")
+        machine2.execute("firewall-cmd --permanent --add-port 49152-49215/tcp")
+        machine2.execute("firewall-cmd --reload")
+
+    def prepareSharedStorage(self, machine1, machine2):
+        # Set up shared NFS storage available on machine1
+        self.restore_file("/etc/exports")
+        nfs_pool = os.path.join(self.vm_tmpdir, "nfs_pool")
+        mnt_exports = os.path.join(self.vm_tmpdir, "mnt_exports")
+
+        # Set Static Ports for NFS v3 on Ubuntu 20.x
+        if machine1.image in ["ubuntu-2004"]:
+            machine1.execute("sed -i 's/--manage-gids/--manage-gids -p 20048/' /etc/default/nfs-kernel-server && systemctl restart nfs-kernel-server")
+
+        machine1.execute("firewall-cmd --zone public --permanent --add-service=nfs && "
+                         "firewall-cmd --zone public --permanent --add-service=mountd && "
+                         "firewall-cmd --zone public --permanent --add-service=rpc-bind && "
+                         "firewall-cmd --reload")
+        machine1.execute("mkdir -p {0} {1} && "
```

Comment:
Command return error code 252, which means firewalld it is not `NOT_RUNNING`. Firewalld is needs to be started by `machine1.execute("systemctl start firewalld")`

Reply:
Fixed

Reply:
Fixed.